scipy/interpolate/_ndgriddata.py

```diff
@@ -92,12 +92,17 @@ class NearestNDInterpolator(NDInterpolatorBase):
     def __init__(self, x, y, rescale=False, tree_options=None):
         NDInterpolatorBase.__init__(self, x, y, rescale=rescale,
                                     need_contiguous=False,
-                                    need_values=False)
+                                    need_values=True)
         if tree_options is None:
             tree_options = dict()
         self.tree = cKDTree(self.points, **tree_options)
         self.values = np.asarray(y)
 
+    def _calculate_triangulation(self):
+        # This function overrides _calculate_triangulation in NDInterpolatorBase,
+        # since triangulation is not needed for NearestNDInterpolator.
+        pass
```

Comment:
So this is a subroutine which attaches a bunch of things to self. What is it supposed to do modify and can it just return whatever it needs to return. Then attaching the values is best done at the call site.

Reply:
Fixed, according to the other comment about `_calculate_triangulation`.